### PR TITLE
Improve linux packaging and installation

### PR DESCRIPTION
**Pull request by Jacalz**
### Description:

This commit does the following things:
- Handle if the distribution uses `/usr/local` or `/usr` when installing data.
- Install icon to `pixmaps` to ensure that the icon is shown correctly from the desktop file.
- Use an extra directory when packing to avoid cluttering the application root dir.
- Remove the temporary file in the end of the packaging.
- Add a Makefile with an install command for users to be able to install packaged applications separately from running `fyne install`.
    
Tested by installing using `fyne install` and by first packaging and then running `sudo make install` from inside the package.

I also added myself to AUTHORS along with ordering it alphabetically.

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Comment by andydotxyz** [comment]
I don't quite follow the Makefile... the result of "fyne package" should be a full app binary that can be copied or uncompressed in place. Install scripts belong in package maintainer helpers I would have thought?

**Comment by Jacalz** [comment]
Yeah absolutely, but like the story about /use and /use/local is an example of when it might not be entirely easy to just copy it into place. While, yes makefiles are generally for packagers, but this way a user can install it much easier than copying folder for folder. Having the makefile gives a user just one command to run and install and application. Not all apps are in repositories so this makes it easier if a user grabs a release from a release page on GitHub for example.

**Comment by andydotxyz** [comment]
The app packages are designed for specific systems, so the package command taking a parameter to determine /usr vs /usr/local would avoid the need for makefiles.
My understanding was that /usr/local is default so that's what we do - if you want to download then install to /usr you can use a temporary space and cp -r mycheckout/usr/local/* /usr

**Comment by Jacalz** [comment]
For future reference: In the talk today, we more or less concluded that the Makefile is a good option for giving users more flexibility of installing packaged applications. The future could see Appimages getting built or something else, but until then this might be a good option.

File: cmd/fyne/package.go [hunk 1]
```diff
@@ -93,42 +93,75 @@ type packager struct {
...
-		"Name="+p.name+"\n"+
-		"Exec="+filepath.Base(p.exe)+"\n"+
-		"Icon="+p.name+"\n")
-	if err != nil {
-		return errors.Wrap(err, "Failed to write desktop entry string")
+	if p.install {
+		tempDir = ""
+	}
+
+	if _, err := os.Stat(filepath.Join("usr", "local")); os.IsNotExist(err) {
```
Comment (andydotxyz) [inline cmd/fyne/package.go:104]: This line does not work on my system - I guess it is rooted at PWD?
The suggestion, however, works.

```suggestion
	if _, err := os.Stat(filepath.Join("/", "usr", "local")); os.IsNotExist(err) {
```

File: cmd/fyne/package.go [hunk 2]
```diff
@@ -93,42 +93,75 @@ type packager struct {
...
+		_, err := io.WriteString(makefile, `ifneq ("$(wildcard /usr/local)","")`+"\n"+
+			"DESTDIR ?= /usr/local\n"+
+			"else\n"+
+			"DESTDIR ?= /usr\n"+
+			"endif\n"+
+			"\n"+
+			"install:\n"+
+			"	install -Dm00644 usr/"+local+"share/applications/"+p.name+`.desktop $(DESTDIR)$(PREFIX)/share/applications/`+p.name+".desktop\n"+
+			"	install -Dm00644 usr/"+local+"bin/"+filepath.Base(p.exe)+` $(DESTDIR)$(PREFIX)/bin`+filepath.Base(p.exe)+"\n"+
+			"	install -Dm00644 usr/"+local+"share/pixmaps/"+p.name+filepath.Ext(p.icon)+` $(DESTDIR)$(PREFIX)/share/icons/hicolor/512x512/apps/`+p.name+filepath.Ext(p.icon)+"\n")
```
Comment (andydotxyz) [inline cmd/fyne/package.go:150]: This destination directory is not the same as is installed by `fyne install`.
Which is wrong?

File: cmd/fyne/package.go [hunk 3]
```diff
@@ -93,42 +93,75 @@ type packager struct {
...
+			return errors.Wrap(err, "Failed to write Makefile string")
+		}
+
+		tarCmd := exec.Command("tar", "zcf", p.name+".tar.gz", "-C", tempDir, "usr", "Makefile")
+		err = tarCmd.Run()
 		if err != nil {
 			return errors.Wrap(err, "Failed to create archive with tar")
 		}
+
+		err = os.RemoveAll(filepath.Join(p.dir, tempDir))
```
Comment (andydotxyz) [inline cmd/fyne/package.go:161]: This is only going to run if there were no errors, is that intended?

File: cmd/fyne/package.go (hunk 2 above)
Comment (Jacalz) [reply cmd/fyne/package.go:150]: Yeah looks like I forgot to update the makefile. Good catch :+1:

File: cmd/fyne/package.go (hunk 1 above)
Comment (Jacalz) [reply cmd/fyne/package.go:104]: Yeah that true. Will get that fixes :)

File: cmd/fyne/package.go (hunk 3 above)
Comment (Jacalz) [reply cmd/fyne/package.go:161]: I didn't think about that. Good point :+1:

**Comment by Jacalz** [comment]
I commited the update during the GitHub server issues. The changes seem to not be visible in this PR. Will create a new one.

**Pull request closed without merging**